website/blog/2024-03-07-using-inner-blocks.md

```diff
@@ -0,0 +1,159 @@
+
+```jsx
+import { InnerBlocks } from '@wordpress/block-editor';
+import { BlockInserter } from '@eightshift/frontend-libs/scripts';
+
+export const CardTwoColumnEditor = ({ attributes, setAttributes, clientId }) => {
+
+	const {
+		cardTwoColumnAllowedBlocks,
+	} = attributes;
+
+	return (
+		<div className={contentClass}>
+			<InnerBlocks
+				allowedBlocks={(typeof cardTwoColumnAllowedBlocks === 'undefined') || cardTwoColumnAllowedBlocks}
+				orientation='vertical'
+				renderAppender={() => <BlockInserter clientId={clientId} />}
+			/>
+		</div>
+	);
+};
+```
+
+The `InnerBlocks` component comes from Gutenberg. In order to limit the allowed blocks, we need to pass the list of allowed inner blocks to the `allowedBlocks` attribute. If your inner blocks should be laid out horizontally in the editor, instead of vertically, you can pass `orientation='horizontal'`.
+
+`renderAppender` prop allows passing a custom block insert button. Due to relatively frequent changes in Gutenberg, a custom `BlockInserter` element is available in Frontend libs as a replacement, and can be passed to `InnerBlocks`.
+
+Finally, to render the inner blocks in PHP, you need to add the following:
+
+```php
+<?php
+// phpcs:ignore Eightshift.Security.ComponentsEscape.OutputNotEscaped
+echo $innerBlockContent;
+?>
+```
+
+And that’s it! Your block now supports inner blocks.
+
+## Overriding inner block attributes
+
+Inner blocks behave like regular blocks, which means they also have a wrapper by default. This may not always be desired and manually removing wrapper or adding custom wrapper settings on each inner block can be tedious. That’s why Eightshift DevKit offers few helpers that allow you to override inner block attributes.
+
+Both of these helpers go to your `{block-name}-block.js` file.
+
+### overrideInnerBlockAttributes
+
+This helper allows you to set the attributes on all inner blocks by passing them as a third parameter. The example below shows how to disable the wrapper and hide wrapper controls.
+
+```jsx
+import { useSelect } from '@wordpress/data';
+import { overrideInnerBlockAttributes } from '@eightshift/frontend-libs/scripts';
+
+export const Accordion = (props) => {
+	// Set these attributes to all inner blocks once inserted in DOM.
+	useSelect((select) => {
+		overrideInnerBlockAttributes(
+			select,
+			props.clientId,
+			{
+				wrapperUse: false,
+				wrapperNoControls: true,
+			},
+		);
+	});
+
+	return (
+		// ...
+	);
+};
+```
+
+Keep in mind this helper allows you to set any attribute that the inner block supports, so you can even set a larger paragraph size and a different color, for example.
+
+### useBlockProps and useInnerBlocksProps
+
+There are also newer inline inner blocks included in WordPress core that allow more control over the InnerBlocks container. Below is the example how to use it.
```

Comment:
```suggestion
In case the default `InnerBlocks` limits your styling and/or layout options too much, you can opt into the newer block prop hooks to implement inner blocks as well!

Import the `useBlockProps` and `useInnerBlocksProps` hooks, configure them just like regular InnerBlocks, and spread the params into an element of your choice (`div` is recommended).

In code, it looks something like this:
```